### ISSUE TITLE
Plugin support

### ISSUE THREAD

**Issue by dogboy23** [drgroot/docker-rtorrent#3]
Would it be possible to install rutorrent plugins? If so, where do you want them to go?

Would be great to have a fully featured version with the simplicity of this container.

Nick

**Comment by dogboy23** [comment]
update... I copied all the existing plugins from [https://github.com/Novik/ruTorrent/tree/master/plugins](https://github.com/Novik/ruTorrent/tree/master/plugins) into /var/www/localhost/htdocs/plugins

I had to install unrar and  ffmpeg:

```
apk add ffmpeg
apk add unrar
rm -rf mediainfo //mediainfo is not available on alpine from what I can gather
```

Also within ~/rtorrent.rc I amended the .session folder to another folder, as one of the plugins needed execute and write permissions. Simplest way, as I don't know much what I am doing.

Nick

**Comment by drgroot** [comment]
It might make more sense to create a volume like /plugins/ that then you can drop standard rutorrent plugins into. 

I kinda like the idea of have a modular plugins folder. The reason why is because more plugins =  more ram required by PHP.

If we have it so the user chooses what plugins are included, then we can still have minimal ram usage and the user can add plugins to fit their needs. I say that because on a synology your already limited on ram. Maintaining a minimal option is kind of ideal.

What plugin requires ffmpeg and unrar?

**Comment by dogboy23** [comment]
this has worked a treat, I am using a volume for the `/.session` folder due to permissions and importing a previous rtorrent torrent list. This worked a treat, I have 19 of 186 torrents with errors [`Download registered as completed, but hash check returned unfinished chunks.`] so I'm just trying to get to the bottom of that.

`unpack` required `unrar`
`screenshots` required `ffmpeg`
`mediainfo` required `mediainfo`
`rutracker_check` required permissions in .session folder

Nick

**Comment by drgroot** [comment]
hmm. 

1. if you remove your plugins, will this work?
2. Does the container have write permissions to the /session mount? 
3. Is the volume exposed as read-only?

**Comment by dogboy23** [comment]
not sure if it was an issue, volume was not read only/permissions were OK.... it worked fine, just those 19 torrents causing problems, I deleted them. All new torrents worked fine.

Unfortunately, I deleted the session by accident, and am now trying to rebuild, without much luck.

**Comment by drgroot** [comment]
hmm. interesting. keep me posted. would like to know the end result. hopefully once i implement the new feature mentined in #6, i will star this one

**Comment by dogboy23** [comment]
all running v. smooth now, very impressed

**Comment by drgroot** [comment]
still a planned feature that I have yet to finish

**Comment by drgroot** [comment]
finally fixed in #18 
However, mediainfo and unpack do not work. Thats because unpack and mediainfo require additional dependencies that I'm trying to avoid to maintain the simplicity of the container.
